python/ray/tune/trainable/metadata.py

```diff
@@ -0,0 +1,96 @@
+import json
+from collections import deque
+from numbers import Number
+from typing import Tuple, Optional
+
+from ray.tune.utils.serialization import TuneFunctionEncoder, TuneFunctionDecoder
+
+
+class _TrainingRunMetadata:
```

Comment:
this is just a lump of some of the fields under `Trial` right? And do we still think everything under `Trial` is driver's view of things?
I feel like putting this under `trainable` folder is kinda misleading or are we planning to have trainable take charge of this eventually?

Reply:
Talked offline, the idea is to eventually have trainables taking care of the management of these data (low priority for 2.7)